### PR TITLE
DDS Visibility of reference API from TopicDescription

### PR DESCRIPTION
**Pull request by IkerLuengo**
Make the reference count API accessible from TopicDescription but hide it from the user on an implementation class

### CODE REVIEW

File: src/cpp/fastdds/topic/TopicDescriptionImpl.hpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    virtual ~TopicDescriptionImpl()
+    {
+    }
+
+    virtual bool is_referenced() const = 0;
+
+    virtual void reference() = 0;
+
+    virtual void dereference() = 0;
```
Comment (MiguelCompany) [inline src/cpp/fastdds/topic/TopicDescriptionImpl.hpp:43]: I would directly implement the logic here, removing it from TopicImpl.

File: include/fastdds/dds/topic/TopicDescription.hpp [hunk 2]
```diff
@@ -26,6 +26,8 @@ namespace fastdds {
 namespace dds {
 
 class DomainParticipant;
+class TopicDescriptionImpl;
+
```
Comment (MiguelCompany) [inline include/fastdds/dds/topic/TopicDescription.hpp:30]:
```suggestion
class TopicDescriptionImpl;
```

File: src/cpp/fastdds/topic/Topic.cpp [hunk 3]
```diff
@@ -105,6 +105,12 @@ ReturnCode_t Topic::get_inconsistent_topic_status(
     return ReturnCode_t::RETCODE_UNSUPPORTED;
 }
 
+TopicDescriptionImpl* Topic::get_impl() const
+{
+    return impl_;
+}
+
+
```
Comment (MiguelCompany) [inline src/cpp/fastdds/topic/Topic.cpp:113]:
```suggestion

```

File: src/cpp/fastdds/topic/TopicDescriptionImpl.hpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+class Topic;
+
+class TopicDescriptionImpl
+{
+
+public:
+
+    TopicDescriptionImpl()
+    : num_refs_(0u)
```
Comment (MiguelCompany) [inline src/cpp/fastdds/topic/TopicDescriptionImpl.hpp:36]:
```suggestion
        : num_refs_(0u)
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1938/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1676/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1932/)

**Pull request merged**